osutil/disks/mockdisk.go

```diff
@@ -62,6 +62,22 @@ type MockDiskMapping struct {
 	DiskSizeInBytes     uint64
 }
 
+func (d *MockDiskMapping) FindMatchingPartitionWithPartUUID(uuid string) (Partition, error) {
+	// TODO: this should just iterate over the static list when that is a thing
+	osutil.MustBeTestBinary("mock disks only to be used in tests")
+
+	for _, p := range d.Structure {
+		if p.FilesystemUUID == uuid {
```

Comment:
shouldn't this use PartitionUUID?

Reply:
Well spotted. I wonder if some tests use the same uuid for the partition than for the filesystem. I will have to figure out why that did not fail.

Reply:
I have found out. For some reason I removed a bit of code and wrote the test slightly work. And that code was not used. Fixed it now.